partner_identification_gln/models/partner.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# ©  2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+from openerp import api, models
+_logger = logging.getLogger(__name__)
+
+try:
+    from stdnum import ean
+    from stdnum.exceptions import InvalidChecksum
+except ImportError:
+    _logger.debug('Cannot `import external dependency python stdnum package`.')
+
+
+class ResPartnerIdCategory(models.Model):
+    _inherit = 'res.partner.id_category'
+
+    @api.multi
+    def validate_res_partner_gln(self, id_number):
+        self.ensure_one()
+        if not id_number:
+            return False
+
+        try:
+            ean.validate(id_number.name)
+        except InvalidChecksum:
+            return True
+
+        cat = self.env.ref('partner_identification_gln.'
+                           'partner_identification_gln_number_category').id
+        num_obj = self.env['res.partner.id_number']
+        duplicate_gln = num_obj.search([('category_id', '=', cat),
+                                        ('name', '=', id_number.name),
+                                        ('name', '!=', False),
+                                        ('id', '!=', id_number.id),
+                                        ('partner_id.active', '=', True)])
```

Comment:
We are not on the base model but on comodel (we don't search on res.partner).
If a partner is inactive, the search will return its numbers. We don't want this behaviour.